### PR TITLE
[Hunter] Tier 29 MM tier

### PR DESCRIPTION
**Pull request by skolldev**
### Description

Adds Tier 29 2p and 4p Statistics for MM. Some minor housekeeping in moving the old SL legendaries to the talent category.

2p should be pretty straightforward, I'd be happy about feedback if I did the calculations for 4p correctly

### Testing

- Test report URL: `/report/L31rt6vbmkJHpQqy/26/118`
- Screenshot(s):
image

### CODE REVIEW

File: src/analysis/retail/hunter/marksmanship/modules/items/T29MMTier4P.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    super(options);
+    this.active = this.selectedCombatant.has4PieceByTier(TIERS.T29);
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.ARCANE_SHOT),
+      this.onShotDamage,
+    );
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.MULTISHOT_MM),
+      this.onShotDamage,
+    );
```
Comment (Pewtro) [inline src/analysis/retail/hunter/marksmanship/modules/items/T29MMTier4P.tsx]:
```suggestion
    this.addEventListener(
      Events.damage.by(SELECTED_PLAYER).spell([SPELLS.ARCANE_SHOT, SPELLS.MULTISHOT_MM]),
      this.onShotDamage,
    );
```

You should be able to do multiple spells in one event listener like so

File: src/analysis/retail/hunter/marksmanship/modules/items/T29MMTier4P.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  statistic() {
+    return (
+      <Statistic category={STATISTIC_CATEGORY.ITEMS} size="flexible">
+        <BoringSpellValueText spell={SPELLS.T29_4P_BONUS_MARKSMANSHIP}>
+          <ItemDamageDone amount={this.totalDamage} />
+          <br />
+          {formatNumber(this.totalProcs)} <small> procs</small>
+          <br />
```
Comment (Pewtro) [inline src/analysis/retail/hunter/marksmanship/modules/items/T29MMTier4P.tsx]: don't need that last breakline

File: src/analysis/retail/hunter/marksmanship/modules/items/T29MMTier4P.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  onShotDamage(event: DamageEvent) {
+    if (!this.hasFocusingAim(event)) {
+      return;
+    }
+    const effectiveDamgeFromCritIncrease = calculateEffectiveDamageFromCritIncrease(
+      event,
+      this.statTracker.currentCritPercentage,
+      1,
```
Comment (Pewtro) [inline src/analysis/retail/hunter/marksmanship/modules/items/T29MMTier4P.tsx]: I'd probably put the 1 in a constant (could be in this file), so we know what it represents if we ever go back and look at this.

**Review by Pewtro — APPROVED** [review]
Minor things but nothing blocking, looks good!

File: src/analysis/retail/hunter/marksmanship/modules/items/T29MMTier4P.tsx (hunk 3 above)
Comment (skolldev) [reply src/analysis/retail/hunter/marksmanship/modules/items/T29MMTier4P.tsx]: Extracted it to a constant 👍

File: src/analysis/retail/hunter/marksmanship/modules/items/T29MMTier4P.tsx (hunk 1 above)
Comment (skolldev) [reply src/analysis/retail/hunter/marksmanship/modules/items/T29MMTier4P.tsx]: That's good to know, thanks!

**Pull request merged**